### PR TITLE
Allowing to break KV foreach loops

### PR DESCRIPTION
**Pull request by rschust**
KV foreach callback functions can now control whether to continue iterating by returning true (continue) or false (stop).

### CODE REVIEW

File: src/kv/kv.h [hunk 1]
```diff
@@ -433,10 +435,10 @@ namespace kv
        * @param F functor, taking a key and a value, return value is ignored
```
Comment (eddyashton) [inline src/kv/kv.h]: This comment is now incorrect.

File: src/kv/kv.h [hunk 2]
```diff
@@ -196,32 +196,34 @@ namespace kv
...
+        [&count](const K& k, const VersionV& v) {
+          count++;
+          return true;
+        });
 
       size_t i = 0;
-      bool ok = true;
-
-      state1.state.foreach([&state2, &ok, &i](const K& k, const VersionV& v) {
+      bool ok = state1.state.foreach([&state2, &ok, &i](const K& k, const VersionV& v) {
```
Comment (jumaffre) [inline src/kv/kv.h]: `ok` should not be captured here

**Comment by jumaffre** [comment]
Should we add a super quick unit test in `kv.h` for this as well?

File: src/kv/kv.h (hunk 1 above)
Comment (rschust) [reply src/kv/kv.h]: Thanks

File: src/kv/kv.h (hunk 2 above)
Comment (rschust) [reply src/kv/kv.h]: Correct, thanks

**Comment by rschust** [comment]
[map_bench_foreach_diff.txt](https://github.com/microsoft/CCF/files/3359336/map_bench_foreach_diff.txt)
@eddyashton @achamayou Looks like a 4%-5% when completing all iterations.

**Pull request merged**